### PR TITLE
roles, services: refactor into modules

### PR DESCRIPTION
**Pull request by zowoq**

### CODE REVIEW

File: build04/configuration.nix [hunk 1]
```diff
@@ -3,15 +3,15 @@
   imports = [
     inputs.disko.nixosModules.disko
     ./hardware-configuration.nix
-    ../roles/common.nix
-    ../roles/hercules-ci
-    ../roles/remote-builder/user.nix
+    inputs.self.modules.nixos.common
```
Comment (Mic92) [inline build04/configuration.nix:6]: Might be worth it to also pass self as a specialArg and use that directly.

File: flake.nix [hunk 2]
```diff
@@ -112,6 +112,23 @@
             };
           };
 
+        flake.modules = {
```
Comment (Mic92) [inline flake.nix:115]: We have a standard attribute for that: https://nixos.wiki/wiki/Flakes#Output_schema

```suggestion
        flake.nixosModules = {
```

**Comment by zowoq** [comment]
Moved the first commit to another PR: https://github.com/nix-community/infra/pull/605

File: build04/configuration.nix (hunk 1 above)
Comment (zowoq) [reply build04/configuration.nix]: I'll leave this for now as we need inputs in a few places anyway.

**Comment by zowoq** [comment]
I'll merge this in 24 hours.

**Review by zimbatm — APPROVED** [review]
LGTM

**Pull request merged**